### PR TITLE
part of scrutinizer fixes

### PR DESCRIPTION
**Pull request by johanderuijter**
Ref: puli/issues#92

### CODE REVIEW

File: src/Util/TwoDimensionalHashMap.php [hunk 1]
```diff
@@ -237,9 +237,11 @@ public function listBySecondaryKey($secondaryKey)
     /**
      * Returns the secondary keys for the given primary key.
      *
-     * @param int|string $primaryKey The primary key.
+     * The primary key is optional. If this argument is not provided, all secondary keys will be returned.
+     *
+     * @param int|string|null $primaryKey The primary key.
      *
-     * @return array The secondary keys.
+     * @return (int|string)[] The secondary keys.
```
Comment (webmozart) [inline src/Util/TwoDimensionalHashMap.php]: Does this notation work? I usually use `int[]|string[]`, even though it's obviously imperfect.

**Comment by webmozart** [comment]
Thanks a lot! :)

File: src/Util/TwoDimensionalHashMap.php (hunk 1 above)
Comment (johanderuijter) [inline src/Util/TwoDimensionalHashMap.php]: I got it from the [PSR-5 Draft](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#arrays), however, considering it's still a draft, i'm actually not a 100% sure.

I can change it to `int[]|string[]` you like that better.

Comment (webmozart) [inline src/Util/TwoDimensionalHashMap.php]: Yes I think that'd be good. We use it in other places and using one consistent way will make it easer to change this everywhere once PSR-5 is finalized.

Comment (johanderuijter) [inline src/Util/TwoDimensionalHashMap.php]: I pushed the change on wednesday, didn't have time and then forgot to notify you about it until now. Let me know if anything else needs to be changed :)

Comment (webmozart) [inline src/Util/TwoDimensionalHashMap.php]: Should be fine, thanks! :)

**Pull request merged**